mantle/kola/harness.go

```diff
@@ -1056,10 +1058,14 @@ func makeNonExclusiveTest(tests []*register.Test, flight platform.Flight) regist
 						defer collectLogsExternalTest(h, t, tcluster)
 					}
 
-					t.Run(tcluster)
+					newTC := cluster.TestCluster{
+						H:       h,
+						Cluster: tcluster.Cluster,
+					}
+					t.Run(newTC)
 				}
 				// Each non-exclusive test is run as a subtest of this wrapper test
-				tcluster.H.Run(t.Name, run)
+				tcluster.H.RunTimeout(t.Name, run, t.Timeout)
```

Comment:
just wondering what the implications of this are for non-exclusive tests. i.e. if I have 40 non-exclusive tests that now run on 1 VM - what is the timeout? The non-exclusive tests are relatively lightweight typically (in my experience) Should we default to something like 1 minute per test (i.e. in this example, 40minutes) or something like that?

Reply:
> if I have 40 non-exclusive tests that now run on 1 VM - what is the timeout?

If the parent test gets stuck here (maybe hangs while bringing up the VM), then the timeout will be 10 mins as long as no subtests have been run yet. In the worst case, if each non-exclusive test hangs then we will have to wait a 10 min timeout for each of the 40 tests (40 x 10min). 

> Should we default to something like 1 minute per test (i.e. in this example, 40minutes) or something like that?

Yep that makes sense, we could do something like that for non-exclusive tests. Some tests that require internet can take a little bit longer (ie. Gursewaks podman test takes ~12s as opposed to misc-ro taking ~2s) , but I think 1 min should still be good.